### PR TITLE
Clippy: Move some attribute lints to be early pass (post expansion)

### PR DESCRIPTION
**Pull request by jdonszelmann**
r? @xFrednet 

As a side effect it removes a duplicated warning on line 53 of the `allow_attributes` test. I discussed this with @xFrednet , and it's mainly to support the attribute rework https://github.com/rust-lang/rust/issues/131229

### CODE REVIEW

**Comment by jdonszelmann** [comment]
ugh forgot to bless the tests, lemme do that

File: src/tools/clippy/clippy_lints/src/attrs/mod.rs [hunk 1]
```diff
@@ -417,12 +417,6 @@ impl_lint_pass!(Attributes => [
     ALLOW_ATTRIBUTES,
     ALLOW_ATTRIBUTES_WITHOUT_REASON,
```
Comment (xFrednet) [inline src/tools/clippy/clippy_lints/src/attrs/mod.rs:418]: NIT, these are now part of the `PostExpansionEarlyAttributes` and are also registered as part of it.

```suggestion
```

File: src/tools/clippy/tests/ui/allow_attributes.stderr [hunk 2]
```diff
@@ -19,13 +19,5 @@ error: #[allow] attribute found
...
 
-error: #[allow] attribute found
-  --> tests/ui/allow_attributes.rs:52:7
-   |
-LL |     #[allow(unused)]
-   |       ^^^^^ help: replace it with: `expect`
-   |
-   = note: duplicate diagnostic emitted due to `-Z deduplicate-diagnostics=no`
-
-error: aborting due to 4 previous errors
```
Comment (xFrednet) [inline src/tools/clippy/tests/ui/allow_attributes.stderr:30]: Something is wrong here. This lint trigger for this has been removed:

```rs
#[clippy::msrv = "1.81"]
fn msrv_1_81() {
    #[allow(unused)]
    let x = 1;
}
```

Same for `allow_attributes_without_reason.rs`. I'm guessing that it has to do with the `clippy::msrv` attribute but I don't understand why. Both this and the old one use `extract_msrv_attr` just with a different context. They should be equivalent, and the MSRV check also looks good... Do you have any idea where this comes from?

**Review by xFrednet — COMMENTED** [review]
Generally LGTM, one incorrect test and a nit, but the rest looks good to me :D

File: src/tools/clippy/tests/ui/allow_attributes.stderr (hunk 2 above)
Comment (jdonszelmann) [reply src/tools/clippy/tests/ui/allow_attributes.stderr:30]: This seems to actually be a bug in the old behaviour – it emitted the same lint twice. I don't know how I fixed it but it seems to be better than before

File: src/tools/clippy/clippy_lints/src/attrs/mod.rs (hunk 1 above)
Comment (jdonszelmann) [reply src/tools/clippy/clippy_lints/src/attrs/mod.rs]: oops, well fixed that.

**Review by xFrednet — APPROVED** [review]

**Comment by xFrednet** [comment]
Excellent, the changes look good to me. One step closer to better attributes :tada: 

---

Roses are red,
Violets are blue,
this only affects clippy,
shall we approve it now?

The changes are steps,
Towards attributes HIR,
Graceful refactoring,
far from MIR

Bors the bot,
waiting for commands,
a rollup I demand

**Pull request merged**